Readme.md

```diff
@@ -0,0 +1,390 @@
+
+A final script was used to retrieve URLs for album images associated with each track in the week. Album images may be more difficult to find then tracks. The script required queries to Spotify, iTunes, and Last.fm's APIs. Each album image found was save to a JSON file mapping track and artist names as keys to the album image URL.
+
+
+#### Sample iTunes Search API Script Snippet
+
+```ruby
+def get_itunes_track(query)
+  res = HTTP.get("https://itunes.apple.com/search?term=#{query}&country=us&limit=1&media=music")
+  if res.code == 200
+    JSON.parse(res)
+  else
+    p "GOT 403"
+    return {"resultCount" => 0}
+  end
+end
+
+def build_query(chart, spotify_id)
+  begin
+    track = JSON.parse(HTTP.get("https://api.spotify.com/v1/tracks/#{spotify_id}").to_s)
+    title = track['name'].gsub(/[^0-9a-zA-Z ]/, "").split(" ").join("+")
+    artist = track['artists'][0]['name'].gsub(/[^0-9a-zA-Z ]/, "").split(" ").join("+");
+  rescue
+    title = chart[0]['title'].gsub(/[^0-9a-zA-Z ]/, "").split(' ').join('+')
+    artist = chart[0]['artist'].gsub(/[^0-9a-zA-Z ]/, "").split(' ').join('+')
+  ensure
+    return [title, artist].join("+");
+  end
+end
+
+trackMeta = {}
+
+charts = JSON.parse(File.read("public/charts/electric/billboard-data.json"))
+
+charts.each do |date, chart|
+  sleep 2
+  p date
+
+    top_song = chart[0]
+    query = build_query(chart, top_song['spotify_id'])
+    response = get_itunes_track(query)
+    i = 1
+    until response["resultCount"] != 0 || i == 10
+      query = build_query(chart, chart[i])
+      response = get_itunes_track(query)
+      i += 1
+    end
+    if (response["resultCount"] > 0)
+      itunes_track = response["results"][0]
+      unless trackMeta[itunes_track['trackId']]
+        trackMeta[date] = { 'previewUrl' => itunes_track['previewUrl'], 'albumImg' => itunes_track['artworkUrl100'] }
+      end
+    end
+end
+
+File.write("public/charts/electric/previewUrls.json", JSON.generate(trackMeta))
+```
+
+### Data Visualization and Graphing
+
+BillboardTopTen takes advantage of React.JS rapid render library for smooth visualization of Billboard's charts. There are two separate React components in charge of data visualization for BillboardTopTen.
+
+The first component is the charts component. This component displays a track's progression over time by drawing out distinct lines following a tracks ranking. The chart component contains an SVG tag split in two five subsection. Each subsection contains ten lines for each track. Every line's vertical coordinate represents a tracks position for the current week while the lines end represents the position of the track in the next week. Each subsection is given a velocity such that the lines are animated across the screen.
```

Comment:
Summarize into one sentence. Mainly just say that we used SVG and combine and summarize the last two sentences.